python/pyspark/ml/tuning.py

```diff
@@ -367,7 +368,9 @@ def copy(self, extra=None):
         """
         if extra is None:
             extra = dict()
-        return CrossValidatorModel(self.bestModel.copy(extra))
+        bestModel = self.bestModel.copy(extra)
+        avgMetrics = [am.copy(extra) for am in self.avgMetrics]
```

Comment:
You're having to convert avgMetrics to a list since it's stored as a numpy.ndarray in CrossValidator.  Could you update CrossValidator itself so that it just uses a list of floats, rather than a numpy array?